core/src/main/java/co/aikar/commands/Annotations.java

```diff
@@ -50,7 +52,7 @@
     }
 
     String getAnnotationValue(AnnotatedElement object, Class<? extends Annotation> annoClass, int options) {
-        Annotation annotation = getAnnotationRecursive(object, annoClass);
+        Annotation annotation = getAnnotationRecursive(object, annoClass, new ArrayList<>()); // TODO: Better collection impl?
```

Comment:
This would be better option `new ConcurrentLinkedQueue<>()`. There is no any ordered objects, just need `add` and `contains` methods so you can use `Queue`.

Reply:
No, this needs to be a HashSet